lib/flat_search_web/live/page_live.ex

```diff
@@ -18,6 +18,73 @@ defmodule FlatSearchWeb.PageLive do
       |> Filters.change_filter(params)
       |> Map.put(:action, :insert)
 
-    {:noreply, assign(socket, changeset: changeset)}
+    params = Enum.map(params, fn {key, value} -> {key, insensitive_string(value)} end)
+
+    {:noreply,
+     assign(socket, changeset: changeset, flats: Flats.get_flats_by(params), filters: params)}
+  end
+
+  @impl true
+  def handle_info({:flat_created, flat}, socket) do
+    case flat_fulfills_filters?(socket.assigns.filters, flat) do
+      true ->
+        {:noreply, assign(socket, flats: [flat] ++ socket.assigns.flats)}
+
+      false ->
+        {:noreply, assign(socket, flats: socket.assigns.flats)}
+    end
+  end
+
+  @impl true
+  def handle_info({:error, reason}, socket) do
+    {:noreply, assign(socket, reason: reason)}
+  end
+
+  defp flat_fulfills_filters?(user_filters, flat_filters) do
+    # Compare only if:
+    # - key in flat_filters exist
+    # - value that key points on in user_filters is not empty
+
+    # support a case for price
+    key_exists_and_has_value =
+      for {key, value} when value not in ["", nil] <- user_filters do
+        case key do
+          "max_price" ->
+            Map.has_key?(flat_filters, :price) and Map.has_key?(flat_filters, :additional_price)
+
+          key ->
+            Map.has_key?(flat_filters, String.to_existing_atom(key))
+        end
+      end
```

Comment:
Can you write this as Enum pipeline?
```
user_filters
|> Enum.reject(fn {_key, value} -> value in ["", nil] end)
|> Enum.map(fn 
 {"max_price", _value} -> ...
 {key, _value} -> ...
end)
```

Reply:
And not sure, but should this use `and` here instead of `or` ->  
`Map.has_key?(flat_filters, :price) and Map.has_key?(flat_filters, :additional_price)`

Reply:
No, in this case they both should be present, later on I add the values together  to check if the sum is greater than max_price wanted by the user